engine/jobmaster/dm/metadata/unit_state.go

```diff
@@ -28,13 +28,14 @@ import (
 
 // UnitState represents the state of units.
 type UnitState struct {
-	state
+	state //nolint:unused
```

Comment:
I forget why we must embed it into a customized State. Can we remove it now? @GMHDBJD

Reply:
`state` is an empty interface. From my understanding, it is not much different from a raw interface.

Reply:
We can remove it.

Reply:
Removed in https://github.com/pingcap/tiflow/pull/7565/commits/ff2ac803e5b92130b448c43c47a29b9b10f3ddb4. @lance6716 @GMHDBJD PTAL again.